src/Engines/thinEngine.ts

```diff
@@ -3361,8 +3400,23 @@ export class ThinEngine {
             }
 
             this._setAnisotropicLevel(this._gl.TEXTURE_3D, texture);
-        }
-        else if (internalTexture && internalTexture.isCube) {
+        } else if (internalTexture && internalTexture.is2DArray) {

```

Comment:
same here, the code is almost the same so jsut detect the texture and use gl.TEXTURE-3D or TEXTURE_2D_ARRAY